data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -487,6 +487,11 @@ public Version getVersionAtSlot(final UInt64 slot) {
     return Version.fromMilestone(spec.atSlot(slot).getMilestone());
   }
 
+  public SafeFuture<Optional<SignedBeaconBlock>> getFinalizedBlockInEffectAtSlot(
+      final UInt64 slot) {
+    return combinedChainDataClient.getFinalizedBlockInEffectAtSlot(slot);
+  }
```

Comment:
so this looks like there's a super simple function here, and instead we've got business logic in an api... 

make this so it returns an optional<Bytes32>, and it either responds with the block root or it responds with empty if the slot isn't finalized or whatever the logic is... but the logic will end up at this point...

Reply:
Done: moved the logic to the ChainDataProvider service